rust_dev_preview/sqs/src/bin/sqs-hello-world.rs

```diff
@@ -14,48 +13,67 @@ struct Opt {
     #[structopt(short, long)]
     region: Option<String>,
 
+    /// Which queue to use. If not provided, uses the first queue found.
+    #[structopt(short, long)]
+    queue: Option<String>,
+
     /// Whether to display additional information.
     #[structopt(short, long)]
     verbose: bool,
 }
 
-// Send a message to and receive a message from a queue.
-// snippet-start:[sqs.rust.sqs-hello-world]
-async fn send_receive(client: &Client) -> Result<(), Error> {
+#[derive(Debug)]
+struct SQSMessage {
+    body: String,
+    group: String,
+}
+
+// snippet-start:[sqs.rust.sqs-list-first]
+async fn find_first_queue(client: &Client) -> Result<String, Error> {
     let queues = client.list_queues().send().await?;
     let queue_urls = queues.queue_urls().unwrap_or_default();
-    let queue_url = match queue_urls.first() {
-        Some(url) => url,
-        None => {
-            eprintln!("No queues in this account. Please create a queue to proceed");
-            exit(1);
-        }
-    };
+    Ok(queue_urls.first().expect("No queues in this account and region. Please create a queue to proceed").to_string())
+}
+// snippet-end:[sqs.rust.sqs-list-first]
 
+// Send a message to a queue.
+// snippet-start:[sqs.rust.sqs-send]
+async fn send(client: &Client, queue_url: &String, message: &SQSMessage) -> Result<(), Error> {
     println!(
-        "Sending and receiving messages on with URL: `{}`",
+        "Sending message to queue with URL: {}",
         queue_url
     );
 
     let rsp = client
         .send_message()
         .queue_url(queue_url)
-        .message_body("hello from my queue")
-        .message_group_id("MyGroup")
+        .message_body(&message.body)
+        .message_group_id(&message.group)
+        // If the queue is fifo, you will need to set .message_deduplication_id
+        // or configure teh queue for ContentBasedDedupliccation
```

Comment:
teh -> the
ContentBasedDedupliccation has an extra 'c' in it
Also, include end punctuation in comments.

Reply:
Done.